### ISSUE TITLE
 Generated TOC does generate nesting

### ISSUE THREAD

**Issue by TheAifam5** [ardumont/markdown-toc#53]
## bug

 Generated TOC does generate nesting.

### Expected behavior

```md
<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
**Table of Contents**

- [Example 1](#example-1)
  - [Content](#content)
- [Example 2](#example-2)
  - [Content](#content-1)

<!-- markdown-toc end -->
```

### Actual behavior

```md
<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
**Table of Contents**

- [Example 1](#example-1)
- [Content](#content)
- [Example 2](#example-2)
- [Content](#content-1)

<!-- markdown-toc end -->
```

### Steps to reproduce the behavior

```md
Title
=====

# Example 1
aa

## Content
xx

# Example 2
yy

## Content 
dd
```

**Comment by dominicm00** [comment]
To make a nested TOC, the buffer-local `imenu-create-index-function` has to be set to `markdown-imenu-create-nested-index` instead of `markdown-imenu-create-flat-index`.

**Comment by goetzc** [comment]
Did this chance with a new recent Emacs version?

**Comment by dominicm00** [comment]
> Did this chance with a new recent Emacs version?

It did; maybe a default value was changed?

**Comment by sochotnicky** [comment]
> I tried setting it like this: `(custom-set-variables '(imenu-create-index-function markdown-imenu-create-nested-index)` but it didn't work. any advice here?

It took a while digging around and this seems related to native compilation as far as I can tell. I have a workaround in my config here:
https://codeberg.org/sochotnicky/dotfiles/src/branch/main/dot_doom.d/config.org#headline-64

Basically - disable native compilation for markdown-mode & add a hook to enable nested TOC generation. Might help you as well?

**Comment by andorsk** [comment]
Good idea, but didn't work unfortunately. This is a bummer, as this is a super useful feature for me.

**Comment by sochotnicky** [comment]
> I tried setting it like this: `(custom-set-variables '(imenu-create-index-function markdown-imenu-create-nested-index)` but it didn't work. any advice here?

To be clear - you need to delete any already natively-compiled markdown-mode files or that first part won't be effective. Just make sure you do that...

**Comment by andorsk** [comment]
I had to restart my emacs, but it worked!  Nice callout @sochotnicky

**Comment by xiliuya** [comment]
Hi, replace this funtion will fix it.

```emacs-lisp
(defun markdown-toc--compute-toc-structure (imenu-index)
  "Given a IMENU-INDEX, compute the TOC structure."
  (--mapcat (markdown-toc--compute-toc-structure-from-level
             (if (eq markdown-toc-user-toc-structure-manipulation-fn 'cdr)
                 -1
               0)
             it)
            imenu-index)) 
```
